### PR TITLE
Doc: Guide to remap plugin API.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
I'm working on some remap plugin implementation and I couldn't find this type of guide information anywhere. The only thing I could find was the function definition page, which was a bit too abbreviated to be really useful if you didn't already know how it worked.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
[approve ci docs]

**Comment by SolidWallOfCode** [comment]
Reworked it a bit, should have dealt with all of @alficles's points.

**Comment by SolidWallOfCode** [comment]
[approve ci docs]

File: doc/developer-guide/plugins/remap-plugins.en.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+Cleanup
+=======
+
+When a new :file:`remap.config` is loaded successfully, the prior configuration is cleaned up. For
+each call to :func:`TSRemapNewInstance` a corresponding call to :func:`TSRemapDeleteInstance` is
+called. The only argument is the invocation instance handle, originally provided by the plugin to
+:func:`TSRemapNewInstance`. This is expected to suffice for the plugin to clean up any rule specific
+data.
+
+As part of the old configuration cleanup :func:`TSRemapConfigReload` is called on the plugins in the
```
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply doc/developer-guide/plugins/remap-plugins.en.rst]: No, I think a comma is not appropriate there.

File: doc/developer-guide/plugins/remap-plugins.en.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+   The old configuration, if any, is still active during the call to :func:`TSRemapConfigReload` and
+   therefore calls to :func:`TSRemapDoRemap` may occur concurrently with that function.
+
+The intention of :func:`TSRemapConfigReload` is to provide for temporary data structures used only
+during configuration loading. These can be created as needed when an invocation instance is loaded
+and used in subsequent invocation instance loading, then cleaned up in :func:`TSRemapConfigReload`.
+
+If no rule uses a plugin, it may be unloaded. In that case :func:`TSRemapDone` is called. This is
+an optional entry point, a plugin is not required to provide it. It is called once per plugin, not
```
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply doc/developer-guide/plugins/remap-plugins.en.rst]: Nah, should just drop the trailing clause.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**